core/src/test/java/org/apache/calcite/rex/RexProgramTest.java

```diff
@@ -3288,6 +3288,57 @@ private void checkSarg(String message, Sarg sarg,
         not(rexBuilder.makeCall(SqlStdOperatorTable.SIMILAR_TO, ref, literal("%"))));
   }
 
+  @Test void testSimplifyNullCheckInFilter() {
```

Comment:
Can we split the test into multiple tests?

Upon failure it should be clear what went wrong, by bundling all in one test it's harder to tell. 

This said, I understand that it will repeat the first part but I still think it's a price worth paying

Reply:
I can do that, but actually it is not very clear for me, which style to follow. I think that majority of the tests in this file are collected to large methods, and also some of them have an honor to reside in their own method.

Are there any guidelines in Calcite regarding this?

Reply:
AFAIK we don't have strong guidelines around this, we mostly act as you did, we look at existing code and we try to adapt to it.

Concerning the two conflicting styles, I see that older tests (> 5 years ago) are more inclined to put lots of cases inside a single test, newer code seems to be more for "single case - single test", which might be a bit more verbose for the "data preparation", but makes tests more readable and debuggable IMO.

I don't feel strongly about this but I favour the "newer" style. If nobody else chimes in and you prefer the current organization I won't object.